streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -738,7 +739,7 @@ public boolean process(final long wallClockTime) {
 
             // after processing this record, if its partition queue's buffered size has been
             // decreased to the threshold, we can then resume the consumption on this partition
-            if (recordInfo.queue().size() == maxBufferedSize) {
+            if (maxBufferedSize != -1 && recordInfo.queue().size() == maxBufferedSize) {
```

Comment:
nit: could we add a TODO here that this logic should be removed once we remove the deprecated old config as well?